test/runnable/literal.d

```diff
@@ -174,16 +174,28 @@ void f13907_3(wchar[3] a) {}
 auto f13907_12(char[1]) { return 1; }
 auto f13907_12(char[2]) { return 2; }
 
+auto f13907_123(char[1]) { return 1; }
+auto f13907_123(char[2]) { return 2; }
+auto f13907_123(char[3]) { return 3; }
+auto f13907_123(const(char)[]) { return 0; }
+
 void test13907()
 {
     static assert(!__traits(compiles, { f13907_1("\U00010000"w); }));
     static assert(!__traits(compiles, { f13907_1("\U00010000" ); }));
     f13907_2("\U00010000"w);
     f13907_2("\U00010000");
-    static assert(!__traits(compiles, { f13907_3("\U00010000"w); }));
-    static assert(!__traits(compiles, { f13907_3("\U00010000" ); }));
+  //f13907_3("\U00010000"w);    // Re-enable implicit length extension, from issue 13999 (unlisted codegen bug)
```

Comment:
The error appears to happen because the string literal has type `wchar[3]` but `se->len` is set to `2`.

`wchar[2]` can be passed in a single 32-bit stack slot, while `wchar[3]` requires two.  As the calling convention uses callee-cleanup, the mismatch in allocated stack space causes a crash when the caller allocates only 4 bytes, and the callee frees 8, then loads the return address from the wrong offset.

I can see a couple of ways to deal with this:
1. Update len when an implicit cast to `T[N]` is performed (I think something similar is done when character width changes?)
2. Change e2ir.c:1277 to pass the static array's dim to `toStringSymbol` instead of `se->len` (there may be other places that need to be updated)

I think 1 is probably the safest option.